gpustat/core.py

```diff
@@ -250,19 +250,38 @@ def jsonify(self):
 
 class GPUStatCollection(object):
 
-    def __init__(self, gpu_list):
+    def __init__(self, gpu_list=[]):
+        """The initialization argument gpu_list is remained to support
+        existing APIs"""
         self.gpus = gpu_list
 
-        # attach additional system information
-        self.hostname = platform.node()
+        N.nvmlInit()
+        device_count = N.nvmlDeviceGetCount()
+        self.handles = [N.nvmlDeviceGetHandleByIndex(idx)
+                        for idx in range(device_count)]
+
+        self.update()
+
+    def __del__(self):
+        # sometimes delayed gc causes problem, just attempt to release
+        # NVML resources
+        try:
+            N.nvmlShutdown()
+        except Exception:
+            pass
+
+    def update(self):
+        self._update_host()
+        self._update_gpu()
+
+    def _update_host(self):
+        """Update additional host information"""
         self.query_time = datetime.now()
+        self.hostname = platform.node()
 
-    @staticmethod
-    def new_query():
```

Comment:
I thought the wrapper function `gpustat.new_query()` was the real API for users, instead of the implementation itself.

It's ok to retain the legacy API, but I prefer a factory function like
```python
@classmethod
def new_query(cls):
    return cls()
```